src/lsp/cobol_config/cobol_config.ml

```diff
@@ -40,16 +40,26 @@ let print_options: Pretty.delayed = fun ppf ->
     ppf !all_configs
 
 let default_search_path =
+  let append root ~sub = match root with
+    | Some d -> [d // sub]
+    | None -> []
+  in
   lazy begin
     let cwd = EzFile.getcwd () in
-    let cob_conf_dir = match Sys.getenv_opt "COB_CONFIG_DIR" with
-      | Some d -> [ d ]
-      | None -> []
-    and xdg_conf_dir = match Sys.getenv_opt "XDG_CONFIG_HOME" with
-      | Some p -> p
-      | None -> Sys.getenv "HOME" // ".config"
+    let cob_config_dir = Option.to_list (Sys.getenv_opt "COB_CONFIG_DIR") in
+    let xdg_superbol_dir =  (* can is this available on win32/cygwin as well? *)
+      append ~sub:"superbol" @@
+      match Sys.getenv_opt "XDG_CONFIG_HOME" with
+      | Some p -> Some p
+      | None -> Option.map (fun d -> d // ".config") (Sys.getenv_opt "HOME")
+    in
+    let system_specific =
+      if Sys.(win32 || cygwin)
+      then append ~sub:"SuperBOL" (Sys.getenv_opt "APPDATA") @
```

Comment:
Does this compile ? I'd expect matching the result of `getenv_opt` with `Some`/`None` before calling `append` ?

Reply:
Ah, forget that, I had not read the code of `append`, which does the match.